cached-fetch/src/stores/createSelfExpiringMemoryStore.ts

```diff
@@ -0,0 +1,42 @@
+import { Store } from "../types";
+
+import { createMemoryStore } from "./createMemoryStore";
+
+const DEFAULT_TTL = 1000;
+
+type Milliseconds = number;
+
+interface Options {
+    ttl?: Milliseconds;
+}
+
+export function createSelfExpiringMemoryStore<T>(options: Options = {}): Store<T> {
+    const store = createMemoryStore<T>()
+    const timers = new Map<string, ReturnType<typeof setTimeout>>()
+    const { ttl = DEFAULT_TTL } = options
+
+    return {
+        has(key) {
+            return store.has(key)
+        },
+        get(key) {
+            return store.get(key)
+        },
+        set(key, value, date?) {
+            clearTimeout(timers.get(key))
+            timers.set(
+                key,
+                setTimeout(() => store.del(key), ttl),
```

Comment:
```suggestion
                setTimeout(() => { store.del(key); clearTimeout(timers.get(key)) }, ttl),
```

i think we need to remove the ref of the timer in the timeout otherwise the happy path would keep the timer ref forever. no?